File: Dart_Source_Files/payload/UserScore.dart
// ignore_for_file: file_names

class UserScore {
  String username;
  int result;
  int? difficulty;
  int? topic;
  int? gameVersion;

  UserScore(
      {required this.username,
      required this.result,
      this.difficulty,
      this.topic,
      this.gameVersion});

  factory UserScore.fromJson(Map<String, dynamic> json) {
    return UserScore(username: json["nickname"], result: json["score"]);
  }

  Map<String, dynamic> toJson() {
    return {
      "username": username,
      "result": result,
      "difficulty": difficulty,
      "topic": topic,
      "gameVersion": gameVersion
    };
  }
}

File: Dart_Source_Files/payload/UserScore.js
class UserScore {

    /**
     * 
     * @param {string} username 
     * @param {number} score 
     * @param {number} difficulty 
     * @param {number} topic 
     * @param {number} gameVersion 
     */
    constructor(username, score, difficulty = undefined, topic = undefined, gameVersion = undefined) {

        this.username = username;
        this.score = score;
        this.difficulty = difficulty;
        this.topic = topic;
        this.gameVersion = gameVersion;
    }

    /**
     * 
     * @param {object} json 
     * @returns {UserScore}
     */
    static fromJson(json) {

        return new UserScore(json.username, json.score);
    }
}

export {UserScore};